modules/core/src/main/java/org/apache/ignite/internal/causality/VersionedValue.java

```diff
@@ -187,34 +189,20 @@ private CompletableFuture<T> getInternal(long causalityToken) {
         long actualToken0 = this.actualToken;
 
         if (history.floorEntry(causalityToken) == null) {
-            throw new OutdatedTokenException(causalityToken, actualToken0, historySize);
+            throw new OutdatedTokenException(causalityToken, actualToken0, maxHistorySize);
         }
```

Comment:
No, it can't because there's an `initFut` variable, so this code will only be executed after history is initialized with something. I believe there's even a test for this situation